src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -2477,6 +2477,55 @@ public IEnumerable<Document> GetRevisionsBinEntries(DocumentsOperationContext co
             }
         }
 
+        public IEnumerable<Document> GetRevisionsBinEntries(DocumentsOperationContext context, DateTime before, DocumentFields fields, RevisionsBinCleanerState state)
+        {
+            var table = new Table(RevisionsSchema, context.Transaction.InnerTransaction);
+
+            var ended = false;
+            while (ended == false)
+            {
+                ended = true;
+
+                using (GetEtagAsSlice(context, state.LastEtag, out var startSlice))
+                {
+                    foreach (var seekResult in table.SeekForwardFrom(RevisionsSchema.Indexes[DeleteRevisionEtagSlice], startSlice, 0))
+                    {
+                        var tvr = seekResult.Result;
+                        var storageId = tvr.Reader.Id;
+                        var etag = TableValueToEtag((int)RevisionsTable.DeletedEtag, ref tvr.Reader);
+                        if (etag == NotDeletedRevisionMarker)
+                        {
+                            context.Transaction.InnerTransaction.ForgetAbout(storageId);
```

Comment:
can you check in other enumerator methods in this class if we are doing this on 'continue' ?

Reply:
On `RevisionsStorage.GetAllRevisions` we do it before we continue..
https://github.com/ravendb/ravendb/blob/6ba5b091fa16246ce5b693ea5aac3d9618f1f700/src/Raven.Server/Documents/Revisions/RevisionsStorage.cs#L1194